lib/routes.dart

```diff
@@ -0,0 +1,92 @@
+import 'package:flutter/material.dart';
+import 'package:gallery/main.dart';
+import 'package:gallery/pages/demo.dart';
+import 'package:gallery/pages/home.dart';
+import 'package:gallery/studies/crane/app.dart';
+import 'package:gallery/studies/fortnightly/app.dart';
+import 'package:gallery/studies/rally/app.dart';
+import 'package:gallery/studies/shrine/app.dart';
+import 'package:gallery/studies/starter/app.dart';
+
+typedef PathWidgetBuilder = Widget Function(BuildContext, Map<String, String>);
+
+class Path {
+  const Path(this.pattern, this.builder);
+
+  /// A RegEx string for route matching.
+  final String pattern;
+
+  /// The builder for the associated pattern route. The first argument is the
+  /// [BuildContext] and the second argument is any RegEx matches if such are
+  /// included inside of the pattern. See example:
+  ///
+  /// ```dart
+  /// Path(
+  ///   'r'^/demo/(?<slug>[\w-]+)$',
+  ///   (context, matches) => Page(argument: matches['slug']),
+  /// )
+  /// ```
+  final PathWidgetBuilder builder;
+}
+
+class RouteConfiguration {
+  /// List of [Path] to for route matching. When a named route is pushed with
+  /// [Navigator.pushNamed], the route name is matched with the [Path.pattern]
+  /// in the list below. As soon as there is a match, the associated builder
+  /// will be returned. This means that the paths higher up in the list will
+  /// take priority.
+  static List<Path> paths = [
+    Path(
+      r'^/demo/(?<slug>[\w-]+)$',
+      (context, matches) => DemoPage(slug: matches['slug']),
+    ),
+    Path(
+      r'^/rally',
+      (context, matches) => StudyWrapper(study: RallyApp()),
+    ),
+    Path(
+      r'^/shrine',
```

Comment:
nit: for cleanliness, I would create these strings using their definitions in the apps, e.g. RallyApp.defaultRoute

Reply:
Sure. Due to to it being RegEx strings I cannot use string interpolation, so it'll be `r'/ + ShrineApp.defaultRoute`